File: packages/agent-sdk-python/src/magicpot_agent_sdk/testing.py
from __future__ import annotations

from collections.abc import Callable
from .contracts import JsonValue


class MemoryAgentTransport:
    def __init__(self, handler: Callable[[str, JsonValue], JsonValue]):
        self._handler = handler
        self.requests: list[tuple[str, JsonValue]] = []

    def request(self, method: str, payload: JsonValue) -> JsonValue:
        self.requests.append((method, payload))
        return self._handler(method, payload)

File: packages/agent-sdk-typescript/src/testing.ts
import type { AgentTransport } from './client.js'
import type { JsonValue } from './contracts.js'

export class MemoryAgentTransport implements AgentTransport {
  readonly requests: Array<{ method: string; payload: JsonValue }> = []

  constructor(
    private readonly handler: (method: string, payload: JsonValue) => JsonValue | Promise<JsonValue>
  ) {}

  async request(method: string, payload: JsonValue): Promise<JsonValue> {
    this.requests.push({ method, payload })
    return this.handler(method, payload)
  }
}
